### PR TITLE
trilinos: v14.4.0 build breaks with cmake@3.31 - so add in a conflict

### PR DESCRIPTION
**Pull request by balay**
Build failure triggered by #47676

https://gitlab.com/xsdk-project/spack-xsdk/-/jobs/8491192131

Fixed with conflict:

https://gitlab.com/xsdk-project/spack-xsdk/-/jobs/8489855021

### CODE REVIEW

File: var/spack/repos/builtin/packages/trilinos/package.py [hunk 1]
```diff
@@ -434,6 +434,7 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("boost+graph+math+exception+stacktrace", when="@:13.4.0 +stk")
     depends_on("cgns", when="+exodus")
     depends_on("cmake@3.23:", type="build", when="@14.0.0:")
+    conflicts("cmake@3.31:", when="@14.4.0")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/trilinos/package.py]: If this is not a regression that will be fixed in the next cmake version, and if older versions of trilinos likely have the same issue, I would suggest adding a [forward compat bound](https://spack.readthedocs.io/en/v0.23.0/packaging_guide.html#specifying-backward-and-forward-compatibility) like this:

```suggestion
    depends_on("cmake@:3.30", type="build", when="@:14.3.0")
```

which has some symmetry with the backward compat bound a line above.

**Comment by haampie** [comment]
Is this a regression in cmake (in which case we'd better open an issue), or is it an intentional breaking change?

File: var/spack/repos/builtin/packages/trilinos/package.py (hunk 1 above)
Comment (balay) [reply var/spack/repos/builtin/packages/trilinos/package.py]: I wasn't sure what range of versions were incompatible - hence limited my initial change to the one version I could verify.

**Comment by balay** [comment]
> Is this a regression in cmake (in which case we'd better open an issue), or is it an intentional breaking change?

I'll defer this to Trilinos developers/maintainers.

File: var/spack/repos/builtin/packages/trilinos/package.py [hunk 2]
```diff
@@ -434,6 +434,7 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("boost+graph+math+exception+stacktrace", when="@:13.4.0 +stk")
     depends_on("cgns", when="+exodus")
     depends_on("cmake@3.23:", type="build", when="@14.0.0:")
+    depends_on("cmake@:3.30", type="build", when="@:14.3.0")
```
Comment (balay) [inline var/spack/repos/builtin/packages/trilinos/package.py]:
```suggestion
    depends_on("cmake@:3.30", type="build", when="@:14.4.0")
```

**Comment by balay** [comment]
closing this PR as this fix is wrong [as I made a mistake in my earlier testing - mixing up 1.0.0 with 1.1.0].

Looks like this build failure is triggered by #47686

**Pull request closed without merging**